### PR TITLE
skeleton test on real redshift server

### PR DESCRIPTION
**Pull request by graingert**
This will probably fail on Travis, as it's completely untested on a real database as of yet!

### CODE REVIEW

**Comment by graingert** [comment]
@jklukas I'm afraid I accidentally leaked the password on Travis and had to change it

**Comment by graingert** [comment]
@jklukas I think these tests will all pass if you give "travis" user 'CREATE DATABASE' permissions

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            conn.execute('CREATE DATABASE {db_name}'.format(db_name=db_name))
+
+        dburl = copy.deepcopy(self.engine.url)
+        dburl.database = db_name
+
+        try:
+            yield db.EngineDefinition(
+                db_connect_url=dburl,
+                connect_args=self.engine_definition.connect_args,
+                poolclass=self.engine_definition.poolclass,
```
Comment (jklukas) [inline tests/conftest.py]: `poolclass` doesn't exist as a parameter in the `EngineDefinition` constructor.

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+@pytest.yield_fixture(scope='function')
+def redshift_session(_session_scoped_redshift_engine):
+    """
+    A redshift session that rolls back all operations.
+
+    The engine and db is maintained for the entire test session for efficiency.
+    """
+    conn = _session_scoped_redshift_engine.connect()
+    tx = conn.begin()
+    session = db.RedshiftSession(bind=conn)
```
Comment (jklukas) [inline tests/conftest.py]: There is no `RedshiftSession` defined in the `db` module.

**Comment by graingert** [comment]
@jklukas cool, thanks for spotting those! Can I get 'CREATE DATABASE' on the redshift cluster?

**Comment by jklukas** [comment]
@graingert : Yeah, the `travis` user now has CREATEDB.

Note that when I tried to run these, I got the following error:
```
E       sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) database "testdb_ae468386480911e587757831c1ccc0fc_0" is being accessed by other users
E        [SQL: 'DROP DATABASE testdb_ae468386480911e587757831c1ccc0fc_0']
```

I'm not sure whether that's going to clear up as we fix other failing tests or not.

**Comment by graingert** [comment]
@jklukas ah you've changed the password again. Do you have some form of IM?

**Comment by jklukas** [comment]
I'm on IRC and Slack. Are you on the #sqlalchemy channel on freenode?

**Comment by jklukas** [comment]
+1

**Pull request merged**